tests/deploy_test.go

```diff
@@ -120,6 +121,156 @@ func Test_Deploy_PassingCustomEnvVars_AndQueryString(t *testing.T) {
 	})
 }
 
+func Test_Auto_Scaling(t *testing.T) {
+	labels := map[string]string{
+		"com.openfaas.scale.min": "1",
+		"com.openfaas.scale.max": "2",
+	}
+
+	functionRequest := requests.CreateFunctionRequest{
+		Image:      "functions/alpine:latest",
+		Service:    "test-auto-scaling",
+		Network:    "func_functions",
+		EnvProcess: "env",
+		Labels:     &labels,
+	}
+
+	deployStatus, deployErr := deploy(t, functionRequest)
+	if deployErr != nil {
+		t.Errorf(deployErr.Error())
+		t.Fail()
+		return
+	}
+
+	if deployStatus != http.StatusOK && deployStatus != http.StatusAccepted {
+		t.Logf("got %d, wanted %d or %d", deployStatus, http.StatusOK, http.StatusAccepted)
+		t.Fail()
+	}
+
+	finishHTTP := make(chan bool)
+
+	go func() {
+		end := false
+		for !end {
+			select {
+			case <-finishHTTP:
+				end = true
+			default:
+				invoke(t, functionRequest.Service, emptyQueryString, http.StatusOK)
+			}
+		}
+	}()
+
+	// calling function for 60 seconds to increase # of replicas
+	endSignal := time.After(60 * time.Second)
+	end := false
+
+	for !end {
+		select {
+		case <-endSignal:
+			t.Logf("expected 2 replicas not 1 after calling function for 60 seconds")
+			t.Fail()
+			finishHTTP <- true
+			end = true
+		default:
+			function := get(t, functionRequest.Service)
+
+			if function.Replicas == 2 {
+				finishHTTP <- true
+				end = true
+			} else {
+				time.Sleep(time.Millisecond * 1000)
+			}
+		}
+	}
+
+	// waiting for replica count to go down
+	endSignal = time.After(60 * time.Second)
```

Comment:
I'd rather not use explicit waits - they are brittle and hard to maintain over time.

Reply:
How should scale up and cool down work without timeouts?

Reply:
What I mean is that 60 seconds is a magic number which may not be correct in all circumstances. It seems like a finger in the air guess which will be brittle over time.

The alternative is to check more than once over a shorter time-frame with a maximum number of attempts.

Reply:
Would you prefer something like the following:

```
const maxAttempts = 60
const minReplicas = 1
i := 0
for ; i < maxAttempts; i++ {
        function := get(t, functionRequest.Service)
	if function.Replicas == minReplicas {
              break;
	}
}

if i == maxAttempts {
  t.Logf("........")
  t.Fail()
}
```